src/PgsqlServer.cpp

```diff
@@ -0,0 +1,222 @@
+
+    _socket.write(type, strlen(type));
+    _socket.write((char*) &message_size, sizeof(message_size));
+    _socket.write(field, strlen(field));
+    _socket.write((char*) delimeter, delim_len);
+    _socket.write(parameter, strlen(parameter));
+}
+
+void PgsqlWorker::sendServerConfiguration(){
+        _socket.write((char*) "R", 1);
+
+        //Not meaning space
+        unsigned char space[] = {0x00, 0x00, 0x00, 0x08, 0x00, 0x00, 0x00, 0x00};
+
+        _socket.write((char*) space,sizeof(space));
+
+        sendParameter("application_name", "", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("client_encoding", "UTF8", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("DateStyle", "ISO, MDY", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("integer_datetimes", "on", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("IntervalStyle", "postgres", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("is_superuser", "on", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("server_encoding", "UTF8", "S");
+        
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("server_version", "9.6.5", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("session_authorization", "postgres", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("standard_conforming_strings", "on", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        sendParameter("TimeZone", "W-SU", "S");
+
+        _socket.write((char*) delimeter, delim_len);
+
+        _socket.write((char*) "K", 1);
+        
+        //Not meaning end of configuration message
+        unsigned char end[] = { 0x00, 0x00, 0x00, 0x0C, 0x00, 0x00, 0x49, 0x4A, 0x01, 0x08, 0x39, 0xFF, 0x5A, 0x00, 0x00, 0x00, 0x05, 0x49};
+    
+        _socket.write((char*) end, sizeof(end));
+}
+
+void PgsqlWorker::run() {
+    size_t size;
+    char key[1];
+    char* received_message;
+
+    size = getMessageSize();
+    received_message = new char[size];
+    _socket.read(received_message, size);
+
+    if(received_message[size-1] == '/') {
+        
+        _socket.write("N", 1);
+
+        size = getMessageSize();
+        received_message = new char[size];
+        _socket.read(received_message, size);
+
+        //Not meaning end 
+        unsigned char answer[13]={0x52, 0x00, 0x00, 0x00, 0x0C, 0x00, 0x00, 0x00, 0x05, 0xDE, 0x35, 0xF9, 0xF9};
+
+        _socket.write((char*) answer, 13);
+
+        _socket.read(key, 1);
+
+        size = getMessageSize();
+        received_message = new char[size];
+        _socket.read(received_message, size);
+        
+        sendServerConfiguration();
+
+    }
+        delete received_message;
+
+        close(sock);
+        exit(1);
```

Comment:
This code just doesn't make sense. You shouldn't use exit() in a thread-based applications. Also sock should be freed in a destructor.